### PR TITLE
Logging and test improvements

### PR DESCRIPTION
**Pull request by rvl**
A few miscellaneous improvements to logging and tests, used for the signTx feature.

### Comments

The individual commit messages should explain things well enough.

### Issue Number

ADP-919

### CODE REVIEW

File: lib/core/src/Cardano/Wallet.hs [hunk 1]
```diff
@@ -2940,30 +2959,63 @@ instance HasSeverityAnnotation WalletLog where
         MsgIsStakeKeyRegistered _ -> Info
 
 data TxSubmitLog
-    = MsgPostTxResult (Hash "Tx") (Either ErrPostTx ())
-    | MsgRetryPostTxResult (Hash "Tx") (Either ErrPostTx ())
+    = MsgSubmitTx Tx TxMeta SealedTx (BracketLog' (Either ErrSubmitTx ()))
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet.hs:2962]: :+1:

File: lib/core/src/Cardano/Wallet/Logging.hs [hunk 2]
```diff
@@ -148,86 +160,187 @@ stdoutTextTracer = Tracer $ liftIO . B8.putStrLn . T.encodeUtf8 . toText
...
+resultSeverity base = \case
+    BracketStart -> base
+    BracketFinish (Left _) -> Error
+    BracketFinish (Right _) -> base
+    BracketException _ -> Error
+    BracketAsyncException _ -> base
+
+{-------------------------------------------------------------------------------
+                             Logging of Exceptions
+-------------------------------------------------------------------------------}
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet/Logging.hs:222]: great stuff!

**Review by paweljakubas — APPROVED** [review]
nice enhancement to logging we have, unifying Wallet.Util (we will probably need to see if more examples of potential `internalError`, etc are in the codebase).

**Comment by rvl** [comment]
Thanks

bors r+

**Review by sevanspowell — APPROVED** [review]

**Pull request merged**